### PR TITLE
chose scorer based on score and fit on model in permutational feature importance

### PR DESCRIPTION
**Pull request by benisraeldan**
#### Reference Issues/PRs

Resolves: #371 

#### What does this implement/fix? Explain your changes.
 
 Scorer can be set on the `model`, but can also be not set so we chose from our defaults based on the type of model

### CODE REVIEW

**Comment by ItayGabbay** [comment]
I feel I can't review it well enough as I'm not so familiar with this part of the code. better @noamzbr @JKL98ISR @matanper will review that

File: deepchecks/utils/features.py [hunk 1]
```diff
@@ -219,13 +220,22 @@ def _calc_importance(
         raise errors.DeepchecksTimeoutError('Permutation importance calculation was not projected to finish in'
                                             f' {permutation_importance_timeout} seconds.')
 
+    if hasattr(model, 'score'):
+        if hasattr(model, 'fit'):
+            scoring = None
```
Comment (matanper) [inline deepchecks/utils/features.py:225]: can you explain why do you check for fit?

Comment (noamzbr) [reply deepchecks/utils/features.py:225]: agree, we don't use fit anywhere

File: deepchecks/utils/features.py [hunk 2]
```diff
@@ -219,13 +220,22 @@ def _calc_importance(
...
+
     r = permutation_importance(
         model,
         dataset_sample.features_columns,
         dataset_sample.label_col,
         n_repeats=n_repeats,
         random_state=random_state,
-        n_jobs=-1
+        n_jobs=-1,
+        scoring=scoring
```
Comment (matanper) [inline deepchecks/utils/features.py:238]: if this is scorer from the model (model.score) it doesn't make sense we did the calculation of time on our default scorer

**Pull request merged**